privacyidea/models.py

```diff
@@ -2699,8 +2699,16 @@ def __init__(self, username, realm, resolver, authentication,
         self.realm = realm
         self.resolver = resolver
         self.authentication = authentication
-        self.first_auth = first_auth
-        self.last_auth = last_auth
+
+        if first_auth:
+            self.first_auth = first_auth
+        else:
+            self.first_auth = datetime.utcnow()
+
+        if last_auth:
+            self.last_auth = last_auth
+        else:
+            self.last_auth = first_auth
```

Comment:
You could write this as
```
self.first_auth = first_auth if first_auth else datetime.utcnow()
self.last_auth = last_auth if last_auth else self.first_auth
```
This is more [*Pythonic*](https://blog.startifact.com/posts/older/what-is-pythonic.html)